internal/controllers/synthesis/pod.go

```diff
@@ -25,42 +27,74 @@ func newPod(cfg *Config, comp *apiv1.Composition, syn *apiv1.Synthesizer) *corev
 		"eno.azure.io/synthesizer-generation": strconv.FormatInt(syn.Generation, 10),
 	}
 
-	userID := int64(1000)
-	yes := true
 	pod.Spec = corev1.PodSpec{
 		ServiceAccountName: cfg.PodServiceAccount,
+		Affinity: &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+					Weight: 100,
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						TopologyKey: "kubernetes.io/hostname",
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								manager.ManagerLabelKey: manager.ManagerLabelValue,
+							},
+						},
+					},
+				}},
+			},
+		},
 		Containers: []corev1.Container{{
 			Name:    "synthesizer",
 			Image:   syn.Spec.Image,
 			Command: []string{"sleep", "infinity"},
 			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: ptr.To(false),
+				ReadOnlyRootFilesystem:   ptr.To(true),
+				RunAsUser:                ptr.To(int64(65532)),
+				RunAsGroup:               ptr.To(int64(65532)),
+				RunAsNonRoot:             ptr.To(true),
 				Capabilities: &corev1.Capabilities{
 					Drop: []corev1.Capability{"ALL"},
 				},
-				RunAsUser:    &userID,
-				RunAsNonRoot: &yes,
-			},
-			Env: []corev1.EnvVar{
```

Comment:
Do you think there's any scenario where this information would be useful within the synth? Either way I think passing this through KRM is a better way of making this available.

Reply:
At least the name of the composition is probably useful - definitely prefer passing it through KRM though. Will make a TODO for this